File: C++/CompilerStudy/Lesson20_Lua3/test/performance.lua

function printPrime(n)
    local start = os.clock()
    for i = 2, n do
        local flag = 1
        for j = 2, i / 2 do
            if i % j == 0 then
                flag = 0
                break
            end
        end
        if flag then end
    end
    print("printPrime: ", os.clock() - start)
end
function perform()
    local start = os.clock()
    for i = 0, 1000000 do end
    print("perform: ", os.clock() - start)
end
function perform_1_5()
    local start = os.clock()
    local i = 0
    while i < 1000000 do i = i + 1 end
    print("perform 1.5: ", os.clock() - start)
end
function perform2()
    local start = os.clock()
    for i = 0, 100 do
        for j = 0, 1000 do
            for k = 0, 1000 do
            end 
        end 
    end
    print("perform2: ", os.clock() - start)
end
function feb(n)
    if n <= 2 then return 1 end
    return feb(n - 1) + feb(n - 2)
end
function perform3()
    local start = os.clock()
    for i = 0, 30 do feb(i) end
    print("perform3: ", os.clock() - start)
end
function main() 
    printPrime(10000)
    perform()
    perform_1_5()
    perform2()
    perform3()
end
main()

File: C++/CompilerStudy/Lesson21_AntlrLang/test/performance.js
var function printPrime(n) {
    var start = clock()
    for (var i = 2; i < n; ++i) {
        var flag = true
        for (var j = 2; j < i / 2; ++j) {
            if (i % j == 0) {
                flag = false
            }
        }
        if (flag) {}
    }
    print('printPrime: ', clock() - start)
}
var function perform() {
    var start = clock()
    for (var i = 0; i < 1000000; ++i);
    print('perform: ', clock() - start)
}
var function perform_1_5() {
    var start = clock()
    var i = 0
    while (i < 1000000) i += 1
    print('perform 1.5: ', clock() - start)
}
var function perform2() {
    var start = clock()
    for (var i = 0; i < 100; ++i) {
        for (var j = 0; j < 1000; ++j) {
            for (var k = 0; k < 1000; ++k) {
            }
        }
    }
    print('perform2: ', clock() - start)
}
var function feb(n) {
    if (n <= 2) return 1
    return feb(n - 1) + feb(n - 2)
}
var function perform3() {
    var start = clock()
    for (var i = 0; i < 30; ++i) feb(i)
    print('perform3: ', clock() - start)
}
var function main()  {
    printPrime(10000)
    perform()
    perform_1_5()
    perform2()
    perform3()
}
main()

